provider/go.mod

```diff
@@ -4,13 +4,13 @@ go 1.18
 
 require (
 	github.com/hashicorp/terraform-provider-vault v1.9.1-0.20221006191932-d603953886dd
```

Comment:
Question from https://github.com/pulumi/pulumi/issues/12300

Are we supposed to bump this? If not, why?

cc @iwahbe @guineveresaenger